# SQL Project (Activsion)
### - Vish Ramesh

## Job Posting <br>
**Company:** Activision Blizzard, Inc. <br>
**Position:** Analytics Internship <br>
<br>
**Qualifications** <br>
• Must be pursuing a bachelors, masters, or PhD majoring in Computer Science, Mathematics, Statistics, Physicsor a related technical field and graduating December 2023 or later <br>
• Curious, dedicated, analytical, and excited to play with data <br>
• Expertise in querying with SQL <br>
• Expertise in quantitative methods – finding the truth and narrative in the data <br>
• Knowledge of data visualization techniques and standard processes <br>
• Programming expertise in Python and/or R is preferable <br>
• Bonus: comfortable working in a Notebook environment (Jupyter, Zeppelin, etc.) <br>

**Responsibilities** <br>
• Explore game or player data in various contexts <br>
• Use modern data science methods to analyze data <br>
• Visualize and interpret analytical results <br>
• Spend a summer focusing on a meaningful project that directly impacts game development while having an unparalleled experience <br>
• Participate in learning and development tech talks to understand how our employees come together to launch AAA games <br>
<br>
**Job description** <br>
At Activision Blizzard we are dedicated to creating the most epic entertainment experiences, driven by our company values and passion, Activision Blizzard creates worlds with the focus to entertain the universe. We are looking for talented and innovative students to join our team for the 2023 summer. All interns will receive a meaningful project and an unparalleled experience. We’ll provide the tools and resources you need to learn. The challenging and creative work you do throughout your internship will directly impact the team you are contributing on.
All interns will be given the choice to work remotely or relocate to the internship location. The company will cover relocation services and will provide a living student to supplement housing, utility, and office supply costs.

# Data Collection

Importing Libraries

In [ ]:
import requests
import json
import pandas as pd
from sqlalchemy import create_engine
from datetime import datetime
import time
import re

# Data Source 1: Steam API - Call of Duty Games

## Table I - List of COD Games (table: steam_gamelist) <br>
Steam API Request (Keyword: Call of Duty) <br>
RapidAPI Link: https://rapidapi.com/psimavel/api/steam2/

### Making the API Request

In [ ]:
# API information
api_key = 'd1bac342f0mshcca7a354cad9ef3p14c9a4jsn3001ee1a3344'

headers = {
    "X-RapidAPI-Host": "steam2.p.rapidapi.com",
    "X-RapidAPI-Key": api_key
}

In [ ]:
base_url = "https://steam2.p.rapidapi.com/search/Call%20of%20Duty/page/"
page = 1

In [ ]:
# an empty list to store API info
all_data = []

**NOTE: it will take 15-20 mins to run the code cell below because of API Limitations**

In [ ]:
# making the API request
while True:
    api_url = base_url + str(page)
    response = requests.get(api_url, headers=headers)

    if response.status_code == 200:
        data = response.json()
        if data:  # Check if the data is not empty
            all_data.extend(data)
            page += 1
            time.sleep(60)  # Add a delay between requests to avoid overloading the API
        else:
            break  # No more data, exit the loop
    else:
        print(f"Error: {response.status_code}")
        break

API Response

In [ ]:
all_data

In [ ]:
type(all_data)

list

### Preparing the Data

In [ ]:
# creating a dict to hold game details
gamelist = {
    'appId': [],
    'title': [],
    'url': [],
    'released': [],
    'price': [],
    'discountedPrice': []
}

# looping through 'all_data' and storing them in the 'gamelist' dict
for game in all_data:
    if 'appId' in game:
        appId = game['appId']
        gamelist['appId'].append(appId)
        print(appId)

        if 'title' in game:
            title = game['title']
            gamelist['title'].append(title)
            print(title)

        if 'url' in game:
            url = game['url']
            gamelist['url'].append(url)
            print(url)

        if 'released' in game:
            try:
                datetime_obj = datetime.strptime(game['released'], '%d %b, %Y')
                released = datetime_obj.strftime('%Y-%m-%d')
            except ValueError:
                try:
                    datetime_obj = datetime.strptime(game['released'], '%b %Y')
                    released = datetime_obj.strftime('%Y-%m')
                except ValueError:
                    released = None
            gamelist['released'].append(released)
            print(released)

        if 'price' in game:
            if 'Free' in game['price']:
                gamelist['price'].append('0')
                gamelist['discountedPrice'].append(None)
                print('0')
            else:
                price_data = re.findall(r'([\d,.]+)€', game['price'])

                if price_data:
                    gamelist['price'].append(price_data[0])
                    print(price_data[0])

                    if len(price_data) > 1:
                        gamelist['discountedPrice'].append(price_data[1])
                        print(price_data[1])
                    else:
                        gamelist['discountedPrice'].append(None)
                else:
                    gamelist['price'].append(None)
                    gamelist['discountedPrice'].append(None)

        print('-' * 100)

In [ ]:
type(gamelist)

dict

In [ ]:
gamelist

### Storing the Data

In [ ]:
# converting gamelist_data dict into a pandas dataframe
df = pd.DataFrame(gamelist)

In [ ]:
df.head()

appId                              title  \
0  1938090  Call of Duty®: Modern Warfare® II   
1  2000950     Call of Duty®: Modern Warfare®   
2   311210       Call of Duty®: Black Ops III   
3   476600                Call of Duty®: WWII   
4  1985820            Call of Duty®: Vanguard   

                                                 url    released  price  \
0  https://store.steampowered.com/app/1938090/Cal...  2022-10-27  69,99   
1  https://store.steampowered.com/app/2000950/Cal...  2023-03-08  59,99   
2  https://store.steampowered.com/app/311210/Call...  2015-11-05  59,99   
3  https://store.steampowered.com/app/476600/Call...  2017-11-02  59,99   
4  https://store.steampowered.com/app/1985820/Cal...  2023-03-08  59,99   

  discountedPrice  
0            None  
1           19,79  
2           19,79  
3           19,79  
4           29,99

In [ ]:
# Remove duplicate appIds
df_unique = df.drop_duplicates(subset=['appId'])

# Display the unique DataFrame
print(df_unique)

Connection to AWS Database and storing the data

In [ ]:
engine = create_engine('mysql+mysqldb://admin:isba_4715@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project?charset=utf8')

In [ ]:
df_unique.to_sql('steam_gamelist', con=engine, if_exists='replace', index=False)

372

# Table II - Game Reviews (table: steam_reviews)

### Collecting the Data (making the request)

In [ ]:
# API information
api_key_2 = 'd1bac342f0mshcca7a354cad9ef3p14c9a4jsn3001ee1a3344'

headers_2 = {
    "X-RapidAPI-Host": "steam2.p.rapidapi.com",
    "X-RapidAPI-Key": api_key_2
}

base_url_2 = 'https://steam2.p.rapidapi.com/appReviews/{}/limit/5/*'

**NOTE: it will take 20 mins to run the code cell below because of API Limitations**

In [ ]:
# all appIds of Call of Duty Games
appIds = [2000950, 1985810, 1985820, 1962663, 1938090, 476600, 393080, 292730, 311210, 209160, 202970, 42680, 42700, 10180, 10090, 7940, 6810, 2630, 2640, 2620]

# an empty dict to store API data
query_summaries = {}

# loop to make the API request for each appId
for appId in appIds:
    api_url_2 = base_url_2.format(appId)
    api_request_2 = requests.get(api_url_2, headers=headers_2)

    if api_request_2.status_code == 200:
        # Process the response data
        data = api_request_2.json()
        query_summary = data.get('query_summary')
        if query_summary:
            query_summaries[appId] = query_summary
        else:
            print(f"No query_summary for appId {appId}")
    else:
        print(f"Error for appId {appId}: {api_request_2.status_code}")

    # Sleep for 60 seconds between each request
    # this is because of the API limit (10 requests per min, but it still kept on crashing for 5/min)
    # so this code take 20 mins to execute (1 request per minute)
    time.sleep(60)

print(query_summaries)

In [ ]:
query_summaries

In [ ]:
type(query_summaries)

dict

### Preparing the Data

In [ ]:
# Initialize an empty dictionary with keys and empty lists as values
review_details = {
    'appId': [],
    'review_score': [],
    'review_score_desc': [],
    'total_positive': [],
    'total_negative': [],
    'total_reviews': []
}

# Loop through the query_summaries dictionary
for appId, details in query_summaries.items():
    # Append the data to the corresponding lists in the review_details dictionary
    review_details['appId'].append(appId)
    review_details['review_score'].append(details['review_score'])
    review_details['review_score_desc'].append(details['review_score_desc'])
    review_details['total_positive'].append(details['total_positive'])
    review_details['total_negative'].append(details['total_negative'])
    review_details['total_reviews'].append(details['total_reviews'])

# Print the review_details dictionary
print(review_details)

In [ ]:
review_details

### Storing the Data

In [ ]:
# converting review_details dict into a pandas dataframe
df_reviews = pd.DataFrame(review_details)

In [ ]:
df_reviews.head()

appId  review_score review_score_desc  total_positive  total_negative  \
0  2000950             6   Mostly Positive             338             137   
1  1985810             8     Very Positive             723             160   
2  1985820             5             Mixed             110              66   
3  1962663             4   Mostly Negative             663            1141   
4  1938090             5             Mixed           71813           41692   

   total_reviews  
0            475  
1            883  
2            176  
3           1804  
4         113505

Connection to AWS Database and storing the data

In [ ]:
engine = create_engine('mysql+mysqldb://admin:isba_4715@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project?charset=utf8')

In [ ]:
df_reviews.to_sql('steam_reviews', con=engine, if_exists='replace', index=False)

20

# Data Source 2: Video Game Sales (kaggle) <br>
**Table:** video_game_sales <br>
**kaggle:** https://www.kaggle.com/datasets/gregorut/videogamesales

In [ ]:
# reading the csv file
file_name = 'vgsales.csv'
df_sales = pd.read_csv(file_name)

In [ ]:
df_sales.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [ ]:
# only including Call of Duty games
cod_df = df_sales[df_sales['Name'].str.contains('Call of Duty')]

In [ ]:
cod_df

Rank                                          Name Platform    Year  \
29        30                Call of Duty: Modern Warfare 3     X360  2011.0   
31        32                       Call of Duty: Black Ops     X360  2010.0   
33        34                     Call of Duty: Black Ops 3      PS4  2015.0   
34        35                    Call of Duty: Black Ops II      PS3  2012.0   
35        36                    Call of Duty: Black Ops II     X360  2012.0   
36        37                Call of Duty: Modern Warfare 2     X360  2009.0   
37        38                Call of Duty: Modern Warfare 3      PS3  2011.0   
40        41                       Call of Duty: Black Ops      PS3  2010.0   
55        56                Call of Duty: Modern Warfare 2      PS3  2009.0   
61        62                          Call of Duty: Ghosts     X360  2013.0   
67        68                          Call of Duty: Ghosts      PS3  2013.0   
70        71                Call of Duty 4: Modern Warfare     X360  2007.0   
93        94                Call of Duty: Advanced Warfare      PS4  2014.0   
98        99                    Call of Duty: World at War     X360  2008.0   
101      102                     Call of Duty: Black Ops 3     XOne  2015.0   
120      121                Call of Duty 4: Modern Warfare      PS3  2007.0   
166      167                    Call of Duty: World at War      PS3  2008.0   
193      194                Call of Duty: Advanced Warfare     XOne  2014.0   
244      245                Call of Duty: Advanced Warfare      PS3  2014.0   
257      258                Call of Duty: Advanced Warfare     X360  2014.0   
349      350                          Call of Duty: Ghosts      PS4  2013.0   
500      501                     Call of Duty: Finest Hour      PS2  2004.0   
505      506                          Call of Duty: Ghosts     XOne  2013.0   
550      551                                Call of Duty 3     X360  2006.0   
711      713                                Call of Duty 3      Wii     NaN   
761      763                Call of Duty: Roads to Victory      PSP  2007.0   
841      843                                Call of Duty 2     X360  2005.0   
861      863                    Call of Duty: World at War      Wii  2008.0   
967      969                     Call of Duty: Black Ops 3      PS3  2015.0   
999     1001                     Call of Duty: Black Ops 3     X360  2015.0   
1035    1037       Call of Duty: World at War Final Fronts      PS2  2008.0   
1037    1039                Call of Duty: Modern Warfare 3       PC  2011.0   
1071    1073          Call of Duty Black Ops: Declassified      PSV  2012.0   
1126    1128                       Call of Duty: Black Ops       PC     NaN   
1217    1219                    Call of Duty: Black Ops II       PC  2012.0   
1223    1225  Call of Duty: Modern Warfare: Reflex Edition      Wii  2009.0   
1238    1240                                Call of Duty 3      PS3  2006.0   
1379    1381                       Call of Duty: Black Ops      Wii  2010.0   
1653    1655                     Call of Duty: Finest Hour       XB  2004.0   
1769    1771                Call of Duty 4: Modern Warfare       PC  2007.0   
1919    1921                                Call of Duty 3      PS2  2006.0   
1944    1946                Call of Duty 4: Modern Warfare       DS  2007.0   
2288    2290                    Call of Duty: World at War       DS  2008.0   
2347    2349                Call of Duty: Modern Warfare 2       PC  2009.0   
2350    2352                Call of Duty: Modern Warfare 3      Wii  2011.0   
2797    2799                          Call of Duty: Ghosts       PC  2013.0   
3211    3213                       Call of Duty: Black Ops       DS  2010.0   
4554    4556                    Call of Duty: Black Ops II     WiiU  2012.0   
4557    4559       Call of Duty: Modern Warfare: Mobilized       DS  2009.0   
4740    4742                Call of Duty: Advanced Warfare

In [ ]:
# creating a connecting to the AWS database
engine = create_engine('mysql+mysqldb://admin:isba_4715@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project?charset=utf8')

In [ ]:
# storing the data
cod_df.to_sql('video_game_sales', con=engine, if_exists='replace', index=False)

59

In [ ]:
# store data for all games (not just COD)
df_sales.to_sql('video_game_sales', con=engine, if_exists='replace', index=False)

16598

# SQL Analysis

In [ ]:
# Load SQL extension
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [ ]:
# Connect to the final)_project database
%sql mysql://admin:isba_4715@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project

In [ ]:
# to install the package
!pip install mysqlclient

## Exploratory Analysis

### 1. To prepare and clean the data

Relevant info from the steam_gamelist table

In [ ]:
%%sql
SELECT appId, title, released, price, discountedPrice
FROM steam_gamelist;

 * mysql://admin:***@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project
372 rows affected.


[('1938090', 'Call of Duty®: Modern Warfare® II', '2022-10-27', '69,99', None),
 ('2000950', 'Call of Duty®: Modern Warfare®', '2023-03-08', '59,99', '19,79'),
 ('311210', 'Call of Duty®: Black Ops III', '2015-11-05', '59,99', '19,79'),
 ('476600', 'Call of Duty®: WWII', '2017-11-02', '59,99', '19,79'),
 ('1985820', 'Call of Duty®: Vanguard', '2023-03-08', '59,99', '29,99'),
 ('202970', 'Call of Duty®: Black Ops II', '2012-11-12', '59,99', '19,79'),
 ('42700', 'Call of Duty®: Black Ops', '2010-11-08', '39,99', '19,99'),
 ('1985810', 'Call of Duty®: Black Ops Cold War', '2023-03-08', '59,99', '19,79'),
 ('292730', 'Call of Duty®: Infinite Warfare', '2016-11-03', '59,99', '19,79'),
 ('209160', 'Call of Duty®: Ghosts', '2014-03-25', '59,99', '19,79'),
 ('2620', 'Call of Duty®', '2006-10-13', '19,99', '9,99'),
 ('2228970', 'Call of Duty®: Modern Warfare® II - BlackCell (Season 03)', '2023-04-12', '29,99', None),
 ('2630', 'Call of Duty® 2', '2006-12-07', '19,99', '14,99'),
 ('2316850', 'Call of Duty®: Modern Warfare® II - Manticore: Pro Pack', '2023-04-20', '19,99', None),
 ('42680', 'Call of Duty®: Modern Warfare® 3', '2011-11-07', '39,99', '19,99'),
 ('10090', 'Call of Duty: World at War', '2008-11-18', '19,99', '14,99'),
 ('2640', 'Call of Duty: United Offensive', '2006-12-07', '19,99', '14,99'),
 ('10180', 'Call of Duty®: Modern Warfare® 2 (2009)', '2009-11-11', '19,99', '14,99'),
 ('2630,2640,2620', 'Call of Duty Warchest', '2006-12-07', '29,99', None),
 ('393080', 'Call of Duty®: Modern Warfare® Remastered (2017)', '2017-07-27', '39,99', '19,99'),
 ('209650', 'Call of Duty®: Advanced Warfare - Gold Edition', '2014-11-02', '59,99', '19,79'),
 ('2137910', 'Call of Duty®: Modern Warfare® II - Urban Veteran: Pro Pack', '2023-02-23', '19,99', None),
 ('2278280', 'Call of Duty®: Modern Warfare® II - Itadakimasu: Starter Pack', '2023-03-23', '9,99', None),
 ('7940', 'Call of Duty® 4: Modern Warfare® (2007)', '2007-11-12', '19,99', '9,99'),
 ('581450', 'Call of Duty®: Black Ops III - Zombies Chronicles', '2017-06-15', '29,99', '9,89'),
 ('2139260', 'Call of Duty League™ - Launch Pack', '2022-11-16', '9,99', None),
 ('259251', 'Call of Duty®: Ghosts - Devastation', '2014-05-08', '14,49', '9,41'),
 ('259250', 'Call of Duty®: Ghosts - Onslaught', '2014-02-27', '14,49', '7,24'),
 ('259252', 'Call of Duty®: Ghosts - Invasion', '2014-07-03', '14,49', '7,24'),
 ('259253', 'Call of Duty®: Ghosts - Nemesis', '2014-09-04', '14,49', '7,24'),
 ('42719', 'Call of Duty®: Black Ops - Rezurrection Content Pack', '2011-09-22', '13,99', '9,09'),
 ('420490', 'Call of Duty®: Infinite Warfare - Season Pass', '2016-11-03', '49,99', '24,99'),
 ('2127790', 'Call of Duty®: Modern Warfare® II - Desert Rogue: Pro Pack', '2022-11-23', '19,99', None),
 ('6810', 'Commandos: Beyond the Call of Duty', '2007-03-15', '4,99', None),
 ('42718', 'Call of Duty®: Black Ops Escalation Content Pack', '2011-06-02', '13,99', '9,09'),
 ('2053671', 'Call of Duty Endowment (C.O.D.E.) - Protector Pack', '2022-10-28', '9,99', None),
 ('683100', 'Call of Duty®: WWII - Season Pass', '2017-11-02', '49,99', '24,99'),
 ('268101', 'Call of Duty®: Ghosts - Wolf Skin', '2014-01-14', '1,99', None),
 ('830450', 'Call of Duty®: Black Ops III - Der Eisendrache Zombies Map', '2018-05-01', '7,99', None),
 ('1962663', 'Call of Duty®: Warzone™ 2.0', '2022-11-16', '0', None),
 ('318792', 'Call of Duty®: Advanced Warfare - Ascendance', '2015-04-30', '14,49', '7,24'),
 ('318793', 'Call of Duty®: Advanced Warfare - Supremacy', '2015-07-02', '14,49', '7,24'),
 ('318791', 'Call of Duty®: Advanced Warfare - Havoc', '2015-03-03', '14,49', '7,24'),
 ('2296290', 'Call of Duty®: Vanguard Points', '2023-03-08', '1,99', None),
 ('219103', 'Call of Duty®: Black Ops II - Apocalypse', '2013-09-26', '14,49', None),
 ('219101', 'Call of Duty®: Black Ops II - Vengeance', '2013-08-01', '14,49', '7,24'),
 ('219098', 'Call of Duty®: Black Ops II - Uprising', '2013-05-16', '14,49', '7,24'),
 ('830461', 'Call of Duty®:

**To only include full games and remove DLCs, packs, bundles, etc. <br>
NOTE: Most further analyses only look at the full individual games.**

In [ ]:
%%sql
SELECT appId, title, released, price, discountedPrice
FROM steam_gamelist
WHERE
    title LIKE '%Call of Duty%'
    AND NOT title LIKE '%-%'
    AND NOT title LIKE '%Points%'
    AND NOT title LIKE '%Pack%'
    AND NOT title LIKE '%Pass%'
    AND NOT title LIKE '%Collection%'
    AND NOT title LIKE '%Warchest%'
ORDER BY released  DESC;

 * mysql://admin:***@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project
20 rows affected.


[('2000950', 'Call of Duty®: Modern Warfare®', '2023-03-08', '59,99', '19,79'),
 ('1985810', 'Call of Duty®: Black Ops Cold War', '2023-03-08', '59,99', '19,79'),
 ('1985820', 'Call of Duty®: Vanguard', '2023-03-08', '59,99', '29,99'),
 ('1962663', 'Call of Duty®: Warzone™ 2.0', '2022-11-16', '0', None),
 ('1938090', 'Call of Duty®: Modern Warfare® II', '2022-10-27', '69,99', None),
 ('476600', 'Call of Duty®: WWII', '2017-11-02', '59,99', '19,79'),
 ('393080', 'Call of Duty®: Modern Warfare® Remastered (2017)', '2017-07-27', '39,99', '19,99'),
 ('292730', 'Call of Duty®: Infinite Warfare', '2016-11-03', '59,99', '19,79'),
 ('311210', 'Call of Duty®: Black Ops III', '2015-11-05', '59,99', '19,79'),
 ('209160', 'Call of Duty®: Ghosts', '2014-03-25', '59,99', '19,79'),
 ('202970', 'Call of Duty®: Black Ops II', '2012-11-12', '59,99', '19,79'),
 ('42680', 'Call of Duty®: Modern Warfare® 3', '2011-11-07', '39,99', '19,99'),
 ('42700', 'Call of Duty®: Black Ops', '2010-11-08', '39,99', '19,99'),
 ('10180', 'Call of Duty®: Modern Warfare® 2 (2009)', '2009-11-11', '19,99', '14,99'),
 ('10090', 'Call of Duty: World at War', '2008-11-18', '19,99', '14,99'),
 ('7940', 'Call of Duty® 4: Modern Warfare® (2007)', '2007-11-12', '19,99', '9,99'),
 ('6810', 'Commandos: Beyond the Call of Duty', '2007-03-15', '4,99', None),
 ('2630', 'Call of Duty® 2', '2006-12-07', '19,99', '14,99'),
 ('2640', 'Call of Duty: United Offensive', '2006-12-07', '19,99', '14,99'),
 ('2620', 'Call of Duty®', '2006-10-13', '19,99', '9,99')]

**INSIGHTS** <br>
1. The steam_gamelist table contains relevant info such as release date, price and the discounted price of each Call of Duty game.
2. The 2nd query filters out all the DLCs, bundles, demos, add-ons, etc. and only includes the individual full games to maintain consistency.

### 2. Basic summary statitics

a) Average price of games in the steam_gamelist table

In [ ]:
%%sql
SELECT AVG(price) AS average_price FROM steam_gamelist;

 * mysql://admin:***@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project
1 rows affected.


[(9.080779944289693,)]

In [ ]:
%%sql
SELECT
    ROUND(AVG(price),2) AS average_price
FROM steam_gamelist
WHERE
    title LIKE '%Call of Duty%'
    AND NOT title LIKE '%-%'
    AND NOT title LIKE '%Points%'
    AND NOT title LIKE '%Pack%'
    AND NOT title LIKE '%Pass%'
    AND NOT title LIKE '%Collection%'
    AND NOT title LIKE '%Warchest%'
    AND price > 0;

 * mysql://admin:***@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project
1 rows affected.


[(40.84,)]

b) Distribution of review scores in the steam_reviews table

In [ ]:
%%sql
SELECT
    review_score,
    COUNT(*) AS count
FROM steam_reviews
GROUP BY review_score
ORDER BY review_score;

 * mysql://admin:***@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project
4 rows affected.


[(4, 1), (5, 6), (6, 1), (8, 12)]

c) Top 10 best-selling games globally in the video_game_sales table

In [ ]:
%%sql
SELECT
    Name,
    Platform,
    Global_Sales
FROM video_game_sales
ORDER BY Global_Sales DESC LIMIT 10;

 * mysql://admin:***@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project
10 rows affected.


[('Wii Sports', 'Wii', 82.74),
 ('Super Mario Bros.', 'NES', 40.24),
 ('Mario Kart Wii', 'Wii', 35.82),
 ('Wii Sports Resort', 'Wii', 33.0),
 ('Pokemon Red/Pokemon Blue', 'GB', 31.37),
 ('Tetris', 'GB', 30.26),
 ('New Super Mario Bros.', 'DS', 30.01),
 ('Wii Play', 'Wii', 29.02),
 ('New Super Mario Bros. Wii', 'Wii', 28.62),
 ('Duck Hunt', 'NES', 28.31)]

**INSIGHTS:**
1. The average price of all the games is €9.08 and €40.84 when not including free games, DLCs, bundles, etc.
3. Based on the review score distribution, most COD games have positive reviews.
4. The top games in video_game sales are all classic nintendo games.

### 3. Relationship between steam_gamelist and steam_reviews table

In [ ]:
%%sql
SELECT
    g.title,
    g.released,
    g.price,
    r.review_score,
    r.total_reviews
FROM steam_gamelist AS g
JOIN steam_reviews AS r
    ON g.appId = r.appId
WHERE
    g.title LIKE '%Call of Duty%'
    AND NOT g.title LIKE '%-%'
    AND NOT g.title LIKE '%Points%'
    AND NOT g.title LIKE '%Pack%'
    AND NOT g.title LIKE '%Pass%'
    AND NOT g.title LIKE '%Collection%'
    AND NOT g.title LIKE '%Warchest%'
ORDER BY
    r.review_score DESC,
    r.total_reviews DESC;

 * mysql://admin:***@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project
20 rows affected.


[('Call of Duty®: Black Ops III', '2015-11-05', '59,99', 8, 74436),
 ('Call of Duty: World at War', '2008-11-18', '19,99', 8, 28488),
 ('Call of Duty®: Modern Warfare® 2 (2009)', '2009-11-11', '19,99', 8, 13284),
 ('Call of Duty®: Black Ops', '2010-11-08', '39,99', 8, 10058),
 ('Call of Duty®: Black Ops II', '2012-11-12', '59,99', 8, 8602),
 ('Call of Duty® 4: Modern Warfare® (2007)', '2007-11-12', '19,99', 8, 8378),
 ('Call of Duty®: Modern Warfare® 3', '2011-11-07', '39,99', 8, 5012),
 ('Call of Duty® 2', '2006-12-07', '19,99', 8, 2875),
 ('Call of Duty®', '2006-10-13', '19,99', 8, 2233),
 ('Call of Duty®: Black Ops Cold War', '2023-03-08', '59,99', 8, 883),
 ('Call of Duty: United Offensive', '2006-12-07', '19,99', 8, 757),
 ('Commandos: Beyond the Call of Duty', '2007-03-15', '4,99', 8, 282),
 ('Call of Duty®: Modern Warfare®', '2023-03-08', '59,99', 6, 475),
 ('Call of Duty®: Modern Warfare® II', '2022-10-27', '69,99', 5, 113505),
 ('Call of Duty®: WWII', '2017-11-02', '59,99', 5, 10821),
 ('Call of Duty®: Infinite Warfare', '2016-11-03', '59,99', 5, 7055),
 ('Call of Duty®: Ghosts', '2014-03-25', '59,99', 5, 5115),
 ('Call of Duty®: Modern Warfare® Remastered (2017)', '2017-07-27', '39,99', 5, 2574),
 ('Call of Duty®: Vanguard', '2023-03-08', '59,99', 5, 176),
 ('Call of Duty®: Warzone™ 2.0', '2022-11-16', '0', 4, 1804)]

**INSIGHT** <br>
The query retrieves information about Call of Duty games, excluding titles containing certain keywords (e.g., "Pack", "Pass", "Collection", etc.). The results show that most Call of Duty games have a review score of 8, with a few exceptions like "Call of Duty®: Modern Warfare®", "Call of Duty®: Modern Warfare® II", and "Call of Duty®: WWII", which have lower scores ranging from 5 to 6. The free-to-play "Call of Duty®: Warzone™ 2.0" has a review score of 4. The titles are ordered by descending review scores and total number of reviews, indicating that "Call of Duty®: Black Ops III" is the highest-rated game with 74,436 reviews.

### 4. Top 5 publishers by global sales (video_game_sales table)

In [ ]:
%%sql
SELECT
    Publisher,
    ROUND(SUM(Global_Sales),2) AS Total_Global_Sales
FROM video_game_sales
GROUP BY Publisher
ORDER BY Total_Global_Sales DESC
LIMIT 5;

 * mysql://admin:***@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project
5 rows affected.


[('Nintendo', 1786.56),
 ('Electronic Arts', 1110.32),
 ('Activision', 727.46),
 ('Sony Computer Entertainment', 607.5),
 ('Ubisoft', 474.72)]

**INSIGHT** <br>
The top 5 publishers by global sales are Nintendo with 1,786.56 million units, Electronic Arts with 1,110.32 million units, Activision with 727.46 million units, Sony Computer Entertainment with 607.5 million units, and Ubisoft with 474.72 million units. Nintendo has the highest total global sales among these publishers, significantly ahead of Electronic Arts and the other publishers in the list. However, Activision demonstrates strong performance in the gaming market.

### 5. Genre-wise sales distribution

In [ ]:
%%sql
SELECT
    Genre,
    ROUND(SUM(Global_Sales), 2) AS global_sales,
    ROUND(SUM(NA_Sales), 2) AS NA_sales,
    ROUND(SUM(EU_Sales), 2) AS EU_sales,
    ROUND(SUM(JP_Sales), 2) AS JP_sales,
    ROUND(SUM(Other_Sales), 2) AS Other_sales
FROM
    video_game_sales
GROUP BY
    Genre
ORDER BY
    global_sales DESC;

 * mysql://admin:***@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project
12 rows affected.


[('Action', 1751.18, 877.83, 525.0, 159.95, 187.38),
 ('Sports', 1330.93, 683.35, 376.85, 135.37, 134.97),
 ('Shooter', 1037.37, 582.6, 313.27, 38.28, 102.69),
 ('Role-Playing', 927.37, 327.28, 188.06, 352.31, 59.61),
 ('Platform', 831.37, 447.05, 201.63, 130.77, 51.59),
 ('Misc', 809.96, 410.24, 215.98, 107.76, 75.32),
 ('Racing', 732.04, 359.42, 238.39, 56.69, 77.27),
 ('Fighting', 448.91, 223.59, 101.32, 87.35, 36.68),
 ('Simulation', 392.2, 183.31, 113.38, 63.7, 31.52),
 ('Puzzle', 244.95, 123.78, 50.78, 57.31, 12.55),
 ('Adventure', 239.04, 105.8, 64.13, 52.07, 16.81),
 ('Strategy', 175.12, 68.7, 45.34, 49.46, 11.36)]

**INSIGHT** <br>
The results show that the Action genre leads with 1,751.18 million units sold, followed by Sports (1,330.93 million), Shooter (1,037.37 million), and Role-Playing (927.37 million) genres. The Shooter genre ranks as the third most popular genre globally, with 1037.37 million units sold. It holds a strong position in North American and European markets with 582.6 and 313.27 million units sold, respectively, while being less prominent in the Japanese market with only 38.28 million units sold. These insights reveal the varying popularity of different video game genres. Activision has managed to capitalize on the high demand for Shooter games.

# Business Problem <br>
**Project Goal:** Examine video game reviews and sales information to generate valuable insights that will assist Activision in expanding its business (boosting sales, profits, user base, etc.). It is essential for game developers to have an in-depth grasp of their users' inclinations.

### Primary Question

**Question:** What Call of Duty games are the most popular, based on percentage of positive reviews? <br>
**Business Justification:** Identifying the most popular Call of Duty games based on positive reviews can inform development and marketing strategies, enhancing user experiences and attracting new customers. This data-driven approach ensures continued franchise success and optimized growth opportunities.

In [ ]:
%%sql
SELECT
    sg.title,
    sr.total_positive,
    sr.total_negative,
    sr.total_reviews,
    ROUND((sr.total_positive * 100.0 / sr.total_reviews),2) AS positive_review_percentage
FROM
    steam_gamelist sg
JOIN
    steam_reviews sr ON sg.appId = sr.appId
WHERE
    sg.title LIKE '%Call of Duty%'
    AND NOT sg.title LIKE '%-%'
    AND NOT sg.title LIKE '%Points%'
    AND NOT sg.title LIKE '%Pack%'
    AND NOT sg.title LIKE '%Pass%'
    AND NOT sg.title LIKE '%Collection%'
    AND NOT sg.title LIKE '%Warchest%'
ORDER BY
    positive_review_percentage DESC;

 * mysql://admin:***@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project
20 rows affected.


[('Call of Duty®', 2111, 122, 2233, Decimal('94.54')),
 ('Call of Duty® 2', 2668, 207, 2875, Decimal('92.80')),
 ('Call of Duty® 4: Modern Warfare® (2007)', 7740, 638, 8378, Decimal('92.38')),
 ('Call of Duty: World at War', 26191, 2297, 28488, Decimal('91.94')),
 ('Call of Duty®: Modern Warfare® 2 (2009)', 12187, 1097, 13284, Decimal('91.74')),
 ('Call of Duty®: Black Ops', 9151, 907, 10058, Decimal('90.98')),
 ('Call of Duty®: Black Ops III', 66047, 8389, 74436, Decimal('88.73')),
 ('Call of Duty: United Offensive', 666, 91, 757, Decimal('87.98')),
 ('Call of Duty®: Modern Warfare® 3', 4377, 635, 5012, Decimal('87.33')),
 ('Commandos: Beyond the Call of Duty', 240, 42, 282, Decimal('85.11')),
 ('Call of Duty®: Black Ops II', 7256, 1346, 8602, Decimal('84.35')),
 ('Call of Duty®: Black Ops Cold War', 723, 160, 883, Decimal('81.88')),
 ('Call of Duty®: Modern Warfare®', 338, 137, 475, Decimal('71.16')),
 ('Call of Duty®: Modern Warfare® II', 71813, 41692, 113505, Decimal('63.27')),
 ('Call of Duty®: Modern Warfare® Remastered (2017)', 1618, 956, 2574, Decimal('62.86')),
 ('Call of Duty®: Vanguard', 110, 66, 176, Decimal('62.50')),
 ('Call of Duty®: WWII', 6644, 4177, 10821, Decimal('61.40')),
 ('Call of Duty®: Infinite Warfare', 3805, 3250, 7055, Decimal('53.93')),
 ('Call of Duty®: Ghosts', 2655, 2460, 5115, Decimal('51.91')),
 ('Call of Duty®: Warzone™ 2.0', 663, 1141, 1804, Decimal('36.75'))]

**Insights:** <br>
1. The top three Call of Duty games with the highest percentage of positive reviews are Call of Duty® (94.54%), Call of Duty® 2 (92.80%), and Call of Duty® 4: Modern Warfare® (2007) (92.38%).
2. Call of Duty®: Modern Warfare® II has a high number of reviews and yet has a lower positive review percentage. <br>

**Recommendations:** <br>
1. Activision should focus on incorporating and promoting the successful elements and features from these top-rated games in future releases, as well as consider remastering or reintroducing these popular titles to attract both existing and new players, ultimately driving sales and user base growth.
2. Analyze user feedback for Call of Duty®: Modern Warfare® II, address identified shortcomings, and make improvements to enhance reception, potentially boosting sales and player engagement.

### Related Questions

**Question 1:** What are the most profitable Activision games both globally and for each region? <br>

**Business Justification:** Identifying the most profitable games globally and regionally enables informed decision-making and resource allocation for game development and marketing. This data-driven approach contributes to the company's overall success and growth in a competitive gaming landscape.

In [ ]:
%%sql
WITH activision_games AS (
    SELECT *
    FROM video_game_sales
    WHERE Publisher = 'Activision'
)
SELECT
    Name,
    ROUND(SUM(Global_Sales),2) AS global_sales,
    ROUND(SUM(NA_Sales),2) AS NA_sales,
    ROUND(SUM(EU_Sales),2) AS EU_sales,
    ROUND(SUM(JP_Sales),2) AS JP_sales,
    ROUND(SUM(Other_Sales),2) AS Other_sales
FROM
    activision_games
GROUP BY
    Name
ORDER BY
global_sales DESC;

 * mysql://admin:***@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project
417 rows affected.


[('Call of Duty: Black Ops', 31.03, 17.59, 9.5, 0.59, 3.36),
 ('Call of Duty: Modern Warfare 3', 30.83, 15.58, 11.29, 0.62, 3.35),
 ('Call of Duty: Black Ops II', 29.72, 14.08, 11.05, 0.72, 3.88),
 ('Call of Duty: Ghosts', 27.38, 14.92, 9.05, 0.48, 2.92),
 ('Call of Duty: Black Ops 3', 25.32, 11.89, 9.56, 0.43, 3.42),
 ('Call of Duty: Modern Warfare 2', 25.09, 13.52, 8.11, 0.46, 3.01),
 ('Call of Duty: Advanced Warfare', 21.9, 10.44, 8.18, 0.34, 2.93),
 ('Call of Duty 4: Modern Warfare', 18.25, 9.97, 5.82, 0.42, 2.05),
 ('Guitar Hero III: Legends of Rock', 16.39, 11.14, 2.59, 0.04, 2.64),
 ('Call of Duty: World at War', 15.87, 9.3, 4.75, 0.0, 1.83),
 ('Destiny', 12.61, 6.59, 4.12, 0.28, 1.62),
 ('Diablo III', 11.08, 4.87, 4.54, 0.21, 1.44),
 ('Guitar Hero: World Tour', 10.05, 6.21, 2.2, 0.0, 1.64),
 ("Tony Hawk's Pro Skater", 8.46, 6.0, 2.16, 0.03, 0.28),
 ('Spider-Man: The Movie', 8.4, 5.56, 2.41, 0.05, 0.37),
 ("Tony Hawk's Pro Skater 3", 8.05, 5.16, 2.29, 0.03, 0.58),
 ('Spider-Man 2', 7.64, 4.74, 2.14, 0.05, 0.69),
 ('Call of Duty 3', 7.56, 4.17, 2.43, 0.05, 0.91),
 ('LEGO Indiana Jones: The Original Adventures', 7.07, 4.26, 2.08, 0.0, 0.75),
 ("Tony Hawk's Pro Skater 2", 6.62, 4.49, 1.88, 0.02, 0.23),
 ("Tony Hawk's Underground", 6.47, 4.11, 1.84, 0.01, 0.5),
 ('Kung Fu Panda', 6.42, 3.42, 1.96, 0.0, 1.02),
 ('World of Warcraft', 6.28, 0.07, 6.21, 0.0, 0.0),
 ('Skylanders SWAP Force', 6.26, 3.53, 2.08, 0.0, 0.66),
 ("Tony Hawk's Pro Skater 4", 5.75, 3.58, 1.75, 0.01, 0.42),
 ('Skylanders Giants', 5.59, 3.01, 1.98, 0.0, 0.6),
 ("Skylanders: Spyro's Adventure", 5.44, 2.69, 2.12, 0.0, 0.63),
 ('LEGO Indiana Jones 2: The Adventure Continues', 5.23, 2.89, 1.72, 0.0, 0.62),
 ('True Crime: Streets of LA', 5.21, 3.29, 1.53, 0.02, 0.38),
 ('Guitar Hero 5', 4.94, 2.4, 1.56, 0.0, 0.97),
 ('StarCraft II: Wings of Liberty', 4.83, 2.56, 1.68, 0.0, 0.59),
 ('Pitfall!', 4.5, 4.21, 0.24, 0.0, 0.05),
 ('Call of Duty: Finest Hour', 4.5, 2.6, 1.6, 0.01, 0.29),
 ('Marvel: Ultimate Alliance', 4.5, 4.02, 0.08, 0.0, 0.4),
 ('Skylanders: Trap Team', 4.4, 2.16, 1.77, 0.0, 0.46),
 ('Guitar Hero: Aerosmith', 4.3, 3.38, 0.43, 0.0, 0.49),
 ('Warcraft II: Tides of Darkness', 4.21, 1.7, 2.27, 0.0, 0.23),
 ('Call Of Duty 2: Big Red One', 4.12, 2.39, 1.39, 0.01, 0.32),
 ("Tony Hawk's Underground 2", 4.11, 2.46, 1.33, 0.01, 0.31),
 ('World of Warcraft: The Burning Crusade', 4.09, 2.57, 1.52, 0.0, 0.0),
 ('007: Quantum of Solace', 3.92, 1.84, 1.35, 0.04, 0.68),
 ('Destiny: The Taken King', 3.87, 1.97, 1.3, 0.06, 0.51),
 ('Spider-Man', 3.66, 2.13, 1.34, 0.02, 0.17),
 ('Guitar Hero: On Tour', 3.48, 2.11, 1.01, 0.01, 0.35),
 ('Madagascar', 3.45, 2.38, 0.92, 0.0, 0.16),
 ('Shrek 2', 3.43, 2.25, 1.08, 0.04, 0.07),
 ('Spider-Man 3', 3.42, 2.35, 0.63, 0.03, 0.42),
 ('Band Hero', 3.4, 1.88, 1.01, 0.0, 0.52),
 ('Transformers: The Game (XBox 360, PS2, PS3, Wii & PC Versions)', 3.37, 2.21, 0.76, 0.01, 0.4),
 ('X-Men Legends', 3.36, 2.03, 1.07, 0.0, 0.26),
 ('DJ Hero', 3.06, 1.55, 0.99, 0.0, 0.52),
 ('Overwatch', 3.01, 1.38, 1.11, 0.14, 0.38),
 ('Bakugan: Battle Brawlers', 2.98, 2.47, 0.26, 0.0, 0.23),
 ('Marvel: Ultimate Alliance 2', 2.85, 1.99, 0.45, 0.0, 0.42),
 ('Guitar Hero: Metallica', 2.74, 1.38, 1.01, 0.0, 0.31),
 ('Transformers: Revenge of the Fallen (XBox 360, PS3, & PC Versions)', 2.74, 1.3, 0.89, 0.0, 0.54),
 ('Guitar Hero: Warriors of Rock', 2.7, 1.28, 1.11, 0.0, 0.31),
 ('Toy Story 2: Buzz Lightyear to the Rescue!', 2.64, 1.7, 0.84, 0.0, 0.09),
 ('[Prototype]', 2.55, 1.49, 0.76, 0.0, 0.31),
 ("Tony Hawk's Project 8", 2.55, 1.87, 0.36, 0.01, 0.33),
 ('The Amazing Spider-Man (Console Version)', 2.23, 1.2, 0.77, 0.0, 0.24),
 ('Skylanders: SuperChargers', 2.23, 1.07, 0.92, 0.0, 0.24),
 ('Guitar Hero II', 2.2, 2.01, 0.02, 0.0, 0.17),
 ('DJ Hero 2', 2.2, 1.6, 0.4, 0.0, 0.19),
 ('Star Wars: Knights of the Old Republic', 2.19, 1.68, 0.44, 0.0, 0.08),
 ('Madagascar: Escape 2 Africa', 2.18, 1.11, 0.62, 0.0, 0.45),
 ('Tony Hawk: RIDE', 2.16, 1.54, 0.42, 0.0, 0.21),


**Insights:** <br>
1. The North American and the European market shows a strong preference for Call of Duty games, with six out of the top ten most profitable games in the region being Call of Duty titles.
2. The Japanese market, however, shows a preference for non-Call of Duty games, with Diablo III (11.08 million units) and Guitar Hero III: Legends of Rock (16.39 million units) being the only Activision games in the top ten.
3. A few of the most popular Call of Duty games on Steam are not the highest selling games.<br>

**Recommendations:** <br>
1. As Call of Duty games have been the most profitable globally, Activision should continue investing in the franchise by prioritizing user feedback, analyzing market trends, and developing strategies that appeal to a broader player base.
2. Activision should also explore opportunities for expanding its market share in regions where Call of Duty games are not as popular (maybe by creating localized content).
3. Conduct further analysis to identify why the most popular Call of Duty games on Steam are not the highest selling, and adjust marketing and game development plans accordingly.

**Question 2:** What is the most popular platform for Call of Duty Games? <br>

**Business Justification:** Knowing the most popular platform for Call of Duty games can help game developers and marketers make strategic decisions regarding game development, marketing campaigns, and distribution channels, which can lead to increased sales and revenue. It can also inform decisions about future game development and investment in emerging platforms.

In [ ]:
%%sql
SELECT Platform,
    ROUND(SUM(Global_Sales),2) AS total_global_sales,
    ROUND(SUM(NA_Sales),2) AS total_na_sales,
    ROUND(SUM(EU_Sales),2) AS total_eu_sales,
    ROUND(SUM(JP_Sales),2) AS total_jp_sales,
    ROUND(SUM(Other_Sales),2) AS total_other_sales
FROM video_game_sales
WHERE Name LIKE 'Call of Duty%'
GROUP BY Platform
ORDER BY total_global_sales DESC;

 * mysql://admin:***@isba-dev-01.c9lqt7cq3kqy.us-east-1.rds.amazonaws.com/final_project
13 rows affected.


[('X360', 94.42, 60.11, 25.5, 0.59, 8.22),
 ('PS3', 80.43, 34.04, 31.27, 2.95, 12.16),
 ('PS4', 25.47, 10.35, 10.53, 0.54, 4.06),
 ('XOne', 15.29, 9.6, 4.39, 0.02, 1.27),
 ('PC', 8.54, 2.02, 5.43, 0.0, 1.08),
 ('PS2', 8.35, 4.49, 2.25, 0.02, 1.59),
 ('Wii', 8.05, 4.76, 2.54, 0.0, 0.76),
 ('DS', 3.33, 2.65, 0.42, 0.01, 0.25),
 ('XB', 2.33, 1.44, 0.8, 0.0, 0.09),
 ('PSP', 2.14, 0.52, 1.03, 0.0, 0.59),
 ('PSV', 1.69, 0.8, 0.52, 0.07, 0.3),
 ('WiiU', 0.79, 0.44, 0.27, 0.01, 0.06),
 ('GC', 0.73, 0.56, 0.15, 0.0, 0.02)]

**Insights:** <br>
1. The Xbox 360 and PlayStation 3 are the most popular platforms for Call of Duty games in terms of global sales, with a combined total of over 174 million units sold. However, it's worth noting that sales on these platforms have been declining in recent years, while sales on the PlayStation 4 and Xbox One have been increasing which makes sense since PS4 and Xbox One are the newer consoles.
2. Although the Xbox 360 has higher global sales compared to the PS3, the PS4 has higher sales compared to the Xbox One, suggesting a potential shift in the popularity of gaming platforms towards PlayStation. <br>

**Recommendations:** <br>
1. Focus on developing and promoting Call of Duty games for the newer platforms, such as the PlayStation 4 and Xbox One, to take advantage of the increasing sales trends. Additionally, developers could consider investing in emerging platforms, such as the Nintendo Switch, to reach new audiences and potentially increase sales.
2. Tailor marketing campaigns and optimize game features specifically for PlayStation platforms to maximize sales and profitability, and continue to monitor sales trends to stay competitive in the market.

## Conclusion <br>
The job posting is for an Analytics Internship at Activision Blizzard, Inc., where interns will work on meaningful projects and gain valuable experience. The SQL analysis of the data provided key insights into the Call of Duty franchise, regional preferences, and gaming platform trends.

Key findings include the top-rated Call of Duty games, the strong preference for Call of Duty games in North American and European markets, and the popularity of different platforms, with a shift towards PlayStation 4 and Xbox One. The analysis also identified discrepancies between popular and highest-selling games on Steam.

Based on these insights, recommendations include focusing on successful elements from top-rated games, addressing shortcomings in games with lower positive review percentages, and investing in the Call of Duty franchise by prioritizing user feedback and market trends. Activision should also explore opportunities to expand its market share in regions with lower Call of Duty popularity and conduct further analysis on Steam's popular games. Finally, the company should focus on developing games for newer platforms, tailoring marketing campaigns, and optimizing game features for PlayStation platforms.